internal/providers/outscale_oapi/utils.go

```diff
@@ -0,0 +1,31 @@
+package outscale_oapi
+
+import (
+	"fmt"
+	"net/http"
+
+	osc "github.com/outscale/osc-sdk-go/v2"
+)
+
+func extractApiError(err error) (bool, *osc.ErrorResponse) {
+	genericError, ok := err.(osc.GenericOpenAPIError)
+	if ok {
+		errorsResponse, ok := genericError.Model().(osc.ErrorResponse)
+		if ok {
+			return true, &errorsResponse
+		}
+		return false, nil
+	}
+	return false, nil
+}
+
+func getErrorInfo(err error, httpRes *http.Response) string {
+	if ok, apiError := extractApiError(err); ok {
+		return fmt.Sprintf("%v %v %v", apiError.GetErrors()[0].GetCode(), apiError.GetErrors()[0].GetType(), apiError.GetErrors()[0].GetDetails())
+	}
```

Comment:
Having internal error details is nice but can you also add the http error code ? This could be useful in case of other errors like 5xx